assets/src/stories-editor/components/editor-carousel/indicator.js

```diff
@@ -29,27 +57,109 @@ const Indicator = ( { pages, currentPage, onClick } ) => {
 	/* translators: %s: Page number */
 	const toolTip = ( pageNumber ) => sprintf( __( 'Go to page %s', 'amp' ), pageNumber );
 
+	const [ isDragging, setIsDragging ] = useState( false );
+	const { movePageToPosition, initializePageOrder } = useDispatch( 'amp/story' );
+	const { getBlockOrder, getBlockIndex } = useSelect( ( select ) => {
+		return select( 'core/block-editor' );
```

Comment:
It's a little simpler and cleaner like this:

```
	const { getBlockOrder, getBlockIndex } = useSelect( ( select ) => select( 'core/block-editor' ) );
```

Actually we often need to just select from a single register, so this might be turned into its own hook to be placed _elsewhere_:

```
const useStore = ( store ) => useSelect( ( select ) => select( store ) );
```

And then this code could be even more simplified to:

```
	const { getBlockOrder, getBlockIndex } = useStore( 'core/block-editor' );
```

Naming to be debated - I think they're called _registries_, but `useRegistry` is already [a low-level hook inside wp.data](https://github.com/WordPress/gutenberg/tree/master/packages/data#useRegistry).